src/adjoint/adjointDebug.F90

```diff
@@ -0,0 +1,736 @@
+            sends: do i=1,commPatternOverset(level, sps)%nProcSend
+            write(*,*) 'commPatternOverset(level, sps)%sendList(i)%interpd ',&
+                        minval(commPatternOverset(level, sps)%sendList(i)%interpd), &
+                        maxval(commPatternOverset(level, sps)%sendList(i)%interpd)
+            end do sends
+            write(*,*) 'internalOverset(level, sps)%donorInterpd ',minval(internalOverset(level, sps)%donorInterpd), &
+                                    maxval(internalOverset(level, sps)%donorInterpd)
+        end if
+
+        write(*,*) 'alphad ',alphad
+        write(*,*) 'betad ',betad
+        write(*,*) 'machd ',machd
+        write(*,*) 'machGridd ',machGridd
+        write(*,*) 'machCoefd ',machCoefd
+        write(*,*) 'pinfdimd ',pinfdimd
+        write(*,*) 'tinfdimd ',tinfdimd
+        write(*,*) 'rhoinfdimd ',rhoinfdimd
+        write(*,*) 'rgasdimd ',rgasdimd
+        write(*,*) 'pointrefd ',pointrefd
+        write(*,*) 'prefd ',prefd
+        write(*,*) 'rhoRefd ',rhoRefd
+        write(*,*) 'Trefd ',Trefd
+        write(*,*) 'murefd ',murefd
+        write(*,*) 'urefd ',urefd
+        write(*,*) 'hrefd ',hrefd
+        write(*,*) 'timerefd ',timerefd
+        write(*,*) 'pinfd ',pinfd
+        write(*,*) 'pinfCorrd ',pinfCorrd
+        write(*,*) 'rhoinfd ',rhoinfd
+        write(*,*) 'uinfd ',uinfd
+        write(*,*) 'rgasd ',rgasd
+        write(*,*) 'muinfd ',muinfd
+        write(*,*) 'gammainfd ',gammainfd
+        write(*,*) 'winfd ',winfd
+        write(*,*) 'veldirfreestreamd ',veldirfreestreamd
+        write(*,*) 'liftdirectiond ',liftdirectiond
+        write(*,*) 'dragdirectiond ',dragdirectiond
+
+        ! Zero all the reverse seeds in the dirichlet input arrays
+        do iDom=1, cgnsNDom
+            do iBoco=1, cgnsDoms(iDom)%nBocos
+            if (associated(cgnsDoms(iDom)%bocoInfo(iBoco)%dataSet)) then
+                do iData=1, size(cgnsDoms(iDom)%bocoInfo(iBoco)%dataSet)
+                    if (associated(cgnsDoms(iDom)%bocoInfo(iBoco)%dataSet(iData)%dirichletArrays)) then
+                        do iDirichlet = 1, size(cgnsDoms(iDom)%bocoInfo(iBoco)%dataSet(iData)%dirichletArrays)
+                        write(*,*) iDom, iBoco, iData, iDirichlet, 'dataArr(:) '&
+                        ,cgnsDomsd(iDom)%bocoInfo(iBoco)%dataSet(iData)%dirichletArrays(iDirichlet)%dataArr(:)
+                        end do
+                    end if
+                end do
+            end if
+            end do
+        end do
+
+        ! And the reverse seeds in the actuator zones
+        do i=1, nActuatorRegions
+            write(*,*) 'actuatorRegionsd(i)%Force ',actuatorRegionsd(i)%force
+            write(*,*) 'actuatorRegionsd(i)%Torque ',actuatorRegionsd(i)%torque
+        end do
+
+    end subroutine printADSeeds
+
+
+#else
+
+    subroutine computeMatrixFreeProductFwdCS(xvdot, extradot, wdot, bcDataValuesdot, &
```

Comment:
Similar with FD, I feel it shall go to adjointAPI.f90.

Reply:
I say we keep it here because that module is not even included in the complex builds whereas we want this to be included.